### PR TITLE
Port UseConditionalExpression analyzer/fixer/tests to shared layer

### PR DESCRIPTION
**Pull request by mavasani**
Recommended reviewing commit by commit

### CODE REVIEW

File: src/Workspaces/Core/Portable/Formatting/Formatter.cs [hunk 1]
```diff
@@ -298,73 +299,6 @@ internal static IList<TextChange> GetFormattedTextChanges(SyntaxNode node, IEnum
                 : formattingResult.GetTextChanges(cancellationToken);
         }
 
-        private static IEnumerable<TextSpan> GetAnnotatedSpans(SyntaxNode node, SyntaxAnnotation annotation)
```
Comment (mavasani) [inline src/Workspaces/Core/Portable/Formatting/Formatter.cs]: Moved to shared `FormattingExtensions` type.

File: src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs [hunk 2]
```diff
@@ -61,28 +58,11 @@ public static IOperation UnwrapImplicitConversion(IOperation value)
                 ? conversion.Operand
                 : value;
 
-        public static SyntaxRemoveOptions GetRemoveOptions(
```
Comment (mavasani) [inline src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs]: Split into separate files - `UseConditionalExpressionHelpers` and `UseConditionalExpressionCodeFixHelpers`

File: src/CodeStyle/Core/Analyzers/Formatting/FormatterHelper.cs [hunk 3]
```diff
@@ -77,6 +78,23 @@ internal static async Task<SyntaxTree> FormatAsync(SyntaxTree syntaxTree, ISynta
...
+        /// Formats the whitespace of a syntax tree.
+        /// </summary>
+        /// <param name="node">The root node of a syntax tree.</param>
+        /// <param name="annotation">The descendant nodes of the root to format.</param>
+        /// <param name="options">An optional set of formatting options. If these options are not supplied the current set of options from the workspace will be used.</param>
+        /// <param name="cancellationToken">An optional cancellation token.</param>
+        /// <returns>The formatted tree's root node.</returns>
+        public static SyntaxNode Format(SyntaxNode node, ISyntaxFormattingService syntaxFormattingService, SyntaxAnnotation annotation, OptionSet options, IEnumerable<AbstractFormattingRule> rules, CancellationToken cancellationToken)
+        {
+            var spans = (annotation == SyntaxAnnotation.ElasticAnnotation)
```
Comment (CyrusNajmabadi) [inline src/CodeStyle/Core/Analyzers/Formatting/FormatterHelper.cs]:
```suggestion
            var spans = annotation == SyntaxAnnotation.ElasticAnnotation
```

File: src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs]: why?

Comment (mavasani) [reply src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs]: SyntaxGenerator/SyntaxRemoveOptions is not available in CodeStyle analyzer layer, so it had to be moved to CodeStyle fixer layer and the rest of the helpers are used by the analyzer, so moved to CodeStyle analyzer layer.

File: src/CodeStyle/Core/Analyzers/Formatting/FormatterHelper.cs (hunk 3 above)
Comment (mavasani) [reply src/CodeStyle/Core/Analyzers/Formatting/FormatterHelper.cs]: This is existing code that I just moved. I am hoping to instead enforce this rule in CI for Workspaces/Features layer once moved to CodeStyle NuGet package.

File: src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Analyzers/Core/Analyzers/UseConditionalExpression/UseConditionalExpressionHelpers.cs]: Thanks!

**Pull request merged**